### PR TITLE
ADEN-12721 Decode adtags on N&R platforms

### PR DESCRIPTION
**Pull request by szymon-nawrocki**
### Ticket
https://fandom.atlassian.net/browse/ADEN-12721

### CODE REVIEW

File: src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts [hunk 1]
```diff
@@ -51,7 +51,10 @@ export class NewsAndRatingsTargetingSetup implements DiProcess {
 			const parts = keyval.split('=');
 
 			if (Array.isArray(parts) && parts.length === 2) {
-				adTagsObj[parts[0]] = parts[1];
+				const key = parts[0];
+				const value = decodeURIComponent(parts[1]).replace(/\+/g, ' ');
```
Comment (nandy-andy) [inline src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts:55]: We don't need a regular expression to replace `'+'` with `' '`, do we? Consider using just `.replace('+', ' ')` 🤔

Comment (szymon-nawrocki) [reply src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts:55]: The value can contain multiple plus signs. The `replace` method will replace only first match

Comment (szymon-nawrocki) [reply src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts:55]: > A string pattern will only be replaced once. To perform a global search and replace, use a regular expression with the g flag, or use [replaceAll()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) instead.

Taken from: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

**Review by nandy-andy — APPROVED** [review]

File: src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts [hunk 2]
```diff
@@ -51,7 +49,10 @@ export class NewsAndRatingsTargetingSetup implements DiProcess {
 			const parts = keyval.split('=');
 
 			if (Array.isArray(parts) && parts.length === 2) {
-				adTagsObj[parts[0]] = parts[1];
+				const key = parts[0];
```
Comment (mfarul) [inline src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts:52]: the whole forEach could be replaced with this:
`for (const [key, value] of new URLSearchParams(adTags)) { 
    adTagsObj[key] = value;
}`
or some map function

**Review by mfarul — CHANGES_REQUESTED** [review]

File: src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts (hunk 2 above)
Comment (szymon-nawrocki) [reply src/platforms/news-and-ratings/shared/setup/context/targeting/news-and-ratings-targeting.setup.ts]: Wow, thanks for the suggestion. Now this complicated splitting and mapping became much more clear one liner

**Review by mfarul — APPROVED** [review]

**Pull request merged**